feature/setting/src/main/java/io/github/droidkaigi/confsched2022/feature/setting/Setting.kt

```diff
@@ -64,45 +71,121 @@ fun Setting(
             verticalArrangement = Arrangement.Top,
             horizontalAlignment = Alignment.Start
         ) {
-            SettingItem.values().forEach { item ->
-                SettingItem(
-                    icon = item.icon,
-                    title = stringResource(item.titleStringRes)
-                )
-            }
+            LanguageSetting()
         }
     }
 }
 
-enum class SettingItem(val titleStringRes: StringResource, val icon: ImageVector) {
-    DarkMode(Strings.setting_item_dark_mode, Icons.Default.DarkMode),
-    Language(Strings.setting_item_language, Icons.Default.Language)
-}
-
 @Composable
-fun SettingItem(
-    icon: ImageVector,
-    title: String,
-    modifier: Modifier = Modifier,
+private fun LanguageSetting(
+    modifier: Modifier = Modifier
 ) {
-    val checkedState = remember { mutableStateOf(true) }
+    val openDialog = remember { mutableStateOf(false) }
+
     Row(
         modifier = modifier
             .padding(vertical = 8.dp)
-            .fillMaxWidth(),
+            .fillMaxWidth()
+            .clickable { openDialog.value = true },
         horizontalArrangement = Arrangement.SpaceBetween,
         verticalAlignment = Alignment.CenterVertically
     ) {
         Row(
             verticalAlignment = Alignment.CenterVertically
         ) {
-            Icon(imageVector = icon, contentDescription = null)
-            Spacer(modifier = Modifier.width(28.dp))
-            Text(text = title)
+            Icon(imageVector = Icons.Default.Language, contentDescription = null)
+            Spacer(modifier = modifier.width(28.dp))
+            Text(text = stringResource(resource = Strings.setting_item_language))
+        }
+    }
+    if (openDialog.value) {
+        LanguageSettingDialog { openDialog.value = false }
+    }
+}
+
+@Composable
+private fun LanguageSettingDialog(onClickConfirm: () -> Unit) {
+    val (selectedLocale, onLocaleSelected) =
```

Comment:
For your information, these destructuring declarations make the recomposition scope wider. Because this read the state. I'll make an issue for this.